### ISSUE TITLE
`RequestCookieJar` duplicates all elements whenever the `CanBuildFrom` is invoked.

### ISSUE THREAD

**Issue by isomarcte** [http4s/http4s#602]
Whenever the `CanBuildFrom` instance for the `RequestCookieJar` is used it duplicates all elements of the current `RequestCookieJar`. This is due to the following definition.

For instance,

```
Welcome to Scala version 2.10.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_66).
Type in expressions to have them evaluated.
Type :help for more information.

scala> import org.http4s._
import org.http4s._

scala> new RequestCookieJar(Seq(new Cookie("foo", "bar")))
res0: org.http4s.RequestCookieJar = RequestCookieJar(foo=bar)

scala> res0 ++ Set()
res1: org.http4s.RequestCookieJar =
RequestCookieJar(foo=bar
foo=bar)

scala>
```

The offending code is here.

```scala
  implicit def canBuildFrom: CanBuildFrom[TraversableOnce[Cookie], Cookie, RequestCookieJar] =
    new CanBuildFrom[TraversableOnce[Cookie], Cookie, RequestCookieJar] {
      def apply(from: TraversableOnce[Cookie]): mutable.Builder[Cookie, RequestCookieJar] =
        newBuilder ++= from
  
      def apply(): mutable.Builder[Cookie, RequestCookieJar] = newBuilder
    }
```

In particular, the the first `apply` method adds all the current elements from `from` into the `newBuilder` instance. This is not required, as the `CanBuildFrom` will already do this. The `from` parameter is not supposed to be used to copy elements from the requesting collections into the builder, but rather to inform what type of builder is to be created, possibly forwarding the creation of the collection type to the `from` parameter.

For our purposes, it doesn't really matter since we want to always use the same `Builder` type regardless of the type of `from`. This is because we care about the element type of the collection, i.e. `Cookie`. This is akin to the [`BitSet`][bitset] type in the standard library (the link is for the factory trait which is responsible for defining the `CanBuildFrom` instance.)

[bitset]: https://github.com/scala/scala/blob/v2.11.8/src/library/scala/collection/generic/BitSetFactory.scala#L35 "bitset"

**Comment by isomarcte** [comment]
Fixed by #603

**Comment by rossabaker** [comment]
Fix will appear in 0.13.2-SNAPSHOT and 0.14.0-SNAPSHOT.
